dts/bindings/mipi-dbi/zephyr,mipi-dbi-spi.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  MIPI-DBI Mode C compatible SPI controller. This driver emulates MIPI DBI
+  mode C using a SPI controller and GPIO pins
+compatible: "zephyr,mipi-dbi-spi"
+
+include: ["mipi-dbi-host.yaml", "spi-controller.yaml", "pinctrl-device.yaml"]
```

Comment:
Why are you  making this an spi-controller? It's not like the driver is then using any SPI api right? I'd expect it to be an SPI device instead and sit under the spi node.

Reply:
The reason I took this approach was that while the MIPI-DBI controller uses a SPI device to implement transactions, it is really closer to a SPI controller than a device IMO. There are a few reasons I think this makes sense:
- properties specific to the display itself (like CPOL/CPHA, SPI frequency, etc) should be set on the display node, not on the MIPI-DBI controller
- Future IP implementing SPI 3/4 wire mode will look a lot like a SPI controller (in fact, I initially considered just implementing that IP as a SPI driver until I realized how poorly it would fit the remainder of the SPI API)

Reply:
I mean you could make this both a controller and device right?

Reply:
Something like this?:
```
&ardunio_spi {
	cs-gpios = <...>;
	
	mipi_dbi_spi@0 {
		reg = <0>;
		compatible = "zephyr,mipi-dbi-spi";
		cd-gpios = <...>;
		
		ili9340@0 {
			reg = <0>;
			compatible = "ilitek, ili9340";
			spi-max-frequency = <..>;
        };
     };
};
```

Then, for future IP that implements MIPI DBI directly we would have something like this:

```
lcd_controller {
	ili9340 {
		reg = <0>;
		compatible = "ilitek, ili9340";
		spi-max-frequency = <..>;
	};
};
```

One thing I think might become a bit confusing for the user is what to use for the `reg` value of the `mipi_dbi_spi` node. That driver does not really need a concept of a SPI slave address, since it will just pass the SPI slave address of the display directly to the SPI driver.

I guess the core question this comes down to is the following: is this `mipi_dbi_spi` driver closer to an emulated controller (like `spi_bitbang`), or closer to a device implementing a protocol using the SPI bus (like `sdhc_spi`). From my point of view, it falls more into the "emulated controller" bucket- but I'm happy to change this if you feel it makes more sense as a device

Reply:
Yeah I think the hierarchy should be like that.

To me, as proposed now, it's an spi based device implementing an spi bus, then if at lower level it does not need one just set `#address-cells = <0>;` and `#size-cells = <0>;` in `zephyr,mipi-dbi-spi` and drop it entirely from the child.

The I guess if you wanted to have multiple displays on the same bus you'd instantiate a `zephyr,mipi-dbi-spi` each.

Reply:
cc @mbolivar-ampere @galak

Reply:
I'm playing with this a bit locally, and I think we could get something like the following to work:

```
&ardunio_spi {
        cs-gpios = <...>;
        
        mipi_dbi_spi@0 {
                reg = <0>;
                compatible = "zephyr,mipi-dbi-spi";
                cd-gpios = <...>;
                #address-cells = <1>;
                #size-cells = <0>;
                
                ili9340@0 {
						reg = <0>;
                        compatible = "ilitek, ili9340";
                        spi-max-frequency = <..>;
                };
        };
};
```

One issue with this- the DTC compiler will flag the `mipi_dbi_spi` node, and complain that it should have the node name "spi". This happens because the `mipi_dbi_spi` node has a register address assigned, which triggers the check in DTC for this warning. We can always silence this warning, but I'm not sure if we want to do that any time the `mipi_dbi_spi` driver is built

Reply:
Sorry- I think I wasn't clear enough above. Specifically, the DTC compiler throws the following warning with a devicetree like the above:

```
/home/danieldegrasse/zephyr/zephyrproject/zephyr/build/zephyr/zephyr.dts:357.16-380.7: Warning (spi_bus_bridge): /soc/peripheral@50000000/flexcomm@123000/mipi_dbi@0: node name for SPI buses should be 'spi'
/dev/null: Warning (spi_bus_reg): Failed prerequisite 'spi_bus_bridge'
```

This is happening because the DTC compiler uses the following heuristic to detect SPI bus nodes that are not named correctly:
- does the child node have properties that start with `spi-`?
- does the parent node have a `reg` property?

You can see the relevant check here: https://github.com/dgibson/dtc/blob/main/checks.c#L1122

this warning won't occur with the existing devicetree definition, because the `mipi_dbi` node doesn't have a `reg` property. In the case where the `mipi_dbi` node is defined as a child of the SPI controller, it must have a `reg` property or the devicetree generation fails.

Reply:
Oh ok I see now. That's annoying, but using the phandle just to work around the warning does not sound great either. :-/ How about making it a "dsi" bus?

Reply:
Unfortunately no, the devicetree tooling will complain :/: 
```
devicetree error: /home/danieldegrasse/zephyr/zephyrproject/zephyr/dts/bindings/mipi-dbi/mipi-dbi-device.yaml (in 'None'): 'on-bus' from included file overwritten ('spi' replaced with 'mipi-dbi')
```

The core issue here is that these displays are in many ways just SPI based displays, with a few extra pins. Because of this, all of the DTS properties for a SPI device make sense within these displays as well. The only reason for the MIPI DBI API to exist is to accommodate "SPI 3/4 wire" display controllers, which don't really fit the bounds of the SPI API.

The way Linux seems to handle this is that MIPI DBI is more of a subsystem than a device driver class. The MIPI DBI API exposes a [mipi_dbi_init](https://github.com/torvalds/linux/blob/master/include/drm/drm_mipi_dbi.h#L154) function, where the display can provide the D/C GPIO pin. However, this won't really work for SPI 3/4 wire display controllers, where a GPIO pin wouldn't be used for the D/C output, and the display controller would need to be initialized as a device.

Reply:
Okay, ah well, good as it is then I guess.

Reply:
Ok, I'll make the updates I mentioned for `rst-invert` and `cd-invert`. Sorry I can't find a better solution here- I agree that the approach right now is very awkward for SPI controllers.